index.bs

```diff
@@ -1188,6 +1188,24 @@ partial interface HTMLIFrameElement {
     Note: This algorithm should be called when a feature policy has
     been <a>violated</a>.
   </section>
+  <section>
+    <h3 id="should-request-be-allowed-to-use-feature">Should <var>request</var> be allowed to use <var>feature</var>?</h3>
+    <p>Given a feature (<var>feature</var>) and a  <a for="/">request</a> (<var>request</var>), this algorithm returns <code>true</code> if the request should be allowed to use <var>feature</var>, and <code>false</code> otherwise.</p>
+    <ol>
+      <li>Set |global object| to |request|’s <a for="request">client</a>’s <a for="environment settings object">global object</a>.</li>
+      <li>Set |document| to null.</li>
+      <li>If |global object| is a {{Document}}, set |document| to |global object|.</li>
+      <li>If |global object| is a {{Window}}, set |document| to |global object|’s <a>associated `Document`</a>.</li>
+      <li>If |document| is null, return <code>false</code>.
+        <div class="issue">How can we enable secure use of policy-controlled features within requests initiated by clients that do not have documents? e.g., from within {{WorkerGlobalScope}}s or {{WorkletGlobalScope}}s, i.e., within Service Workers?</div>
```

Comment:
ServiceWorker's client has an associated origin, we should be checking against that?

@jakearchibald any guidance? :)

Reply:
The CSP model would probably work well here. https://www.w3.org/TR/CSP3/#html-integration

Documents and worker globals have the CSP list. Requests are blocked before reaching the service worker if they fail the document's CSP. However, the service worker may also have its own CSP which blocks additional requests.

CSP also checks responses. I don't know if that's desirable here.

Reply:
Workers having their own Feature Policy is being figured out here: https://github.com/WICG/feature-policy/issues/207. Until that's resolved, I think we can/should just return false in Worker contexts.